### PR TITLE
west_commands: pyocd: issue a warning when falling back to binary file

### PR DESCRIPTION
**Pull request by SIGSTACKFAULT**
Signed-off-by: Connor King <SIGSTACKFAULT@gmail.com>

### CODE REVIEW

File: scripts/west_commands/runners/pyocd.py [hunk 1]
```diff
@@ -111,15 +111,20 @@ def do_run(self, command, **kwargs):
...
-        elif os.path.isfile(self.bin_name):
-            fname = self.bin_name
-        else:
-            raise ValueError(
-                'Cannot flash; no hex ({}) or bin ({}) files'.format(
-                    self.hex_name, self.bin_name))
+            return self.hex_name
+
+        if os.path.isfile(self.bin_name):
+            self.logger.warning('Hex file "{}" not found, falling back to binary "{}"'.format(self.hex_name, self.bin_name))
```
Comment (mbolivar) [inline scripts/west_commands/runners/pyocd.py]: Please keep this file 80-column clean.

File: scripts/west_commands/runners/pyocd.py [hunk 2]
```diff
@@ -132,7 +137,7 @@ def flash(self, **kwargs):
                self.flash_extra +
                [fname])
 
-        self.logger.info('Flashing Target Device')
+        self.logger.info('Flashing Target Device ({})'.format(os.path.basename(fname)))
```
Comment (mbolivar) [inline scripts/west_commands/runners/pyocd.py]: Here too.

**Review by mbolivar — CHANGES_REQUESTED** [review]
Looks good, just some whitespace nits. Fine for 2.0 in my opinion with those fixed.

**Comment by SIGSTACKFAULT** [comment]
> 1: UC2 Signed-off-by: must have a full name

yes it does!

File: scripts/west_commands/runners/pyocd.py (hunk 1 above)
Comment (mbolivar) [reply scripts/west_commands/runners/pyocd.py]: Why was this comment marked resolved? Did you forget to push an update?

**Comment by mbolivar** [comment]
>>    1: UC2 Signed-off-by: must have a full name
> yes it does!

My guess is the "SIGSTACKFAULT" is causing the issue; try removing that maybe?

File: scripts/west_commands/runners/pyocd.py (hunk 1 above)
Comment (SIGSTACKFAULT) [reply scripts/west_commands/runners/pyocd.py]: I have no idea what I'm doing. Never used this feature.

Comment (SIGSTACKFAULT) [reply scripts/west_commands/runners/pyocd.py]: Thought resolving would apply the suggestion

**Comment by SIGSTACKFAULT** [comment]
gotta wait for all the checks to pass before I can read gitlint's output, grumble grumble.

**Comment by SIGSTACKFAULT** [comment]
@mbolivar I've fixed the line length stuff.

**Comment by mbolivar** [comment]
Looks like you need a rebase now that https://github.com/zephyrproject-rtos/zephyr/pull/18310 is merged

**Comment by mbolivar** [comment]
> gotta wait for all the checks to pass before I can read gitlint's output, grumble grumble.

You can run it locally. See the Contribution Guidelines in the documentation.

**Comment by mbolivar** [comment]
@sigstackfault ping

**Comment by SIGSTACKFAULT** [comment]
Welp, I guess this is just redundant.
I should check my email more often.

**Pull request closed without merging**